pyproject.toml

```diff
@@ -1,8 +1,18 @@
 [build-system]
 build-backend = "mesonpy"
+backend-path = ['./vendored-meson/meson-python']
 requires = [
     "Cython>=3.0",
-    "meson-python>=0.13.1",
+    # All dependencies of the vendored meson-python (except for meson, because
+    # we've got that vendored too - that's the point of this exercise).
+    'pyproject-metadata >= 0.7.1',
+    'tomli >= 1.0.0; python_version < "3.11"',
+    'setuptools >= 60.0; python_version >= "3.12"',
```

Comment:
A more recent version of setuptools is also needed for Python 3.11 due to deprecations. I'm running 68.0.0 at the moment. There may be a sweet spot in between, but I don't know what it is.

Reply:
Looks like 62.3.0 is the first version to have the fixes for 3.11, might go later just to be safe.